### PR TITLE
Update artifact repository download methods to return absolute paths

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?
 
Update `ArtifactRepository.download_artifacts()` to always return an absolute filesystem path to the downloaded artifacts.
 
## How is this patch tested?
 
A test case was added to the suite of **local artifact repository tests**. While it seems reasonable to test each artifact repository implementation, @sueann and I agree that it's better to unify test coverage by creating a single test suite that runs against artifact repository implementations; we can prioritize this after the next release.
 
## Release Notes

Fix a bug where relative paths were returned by implementations of `ArtifactRepository.download_artifacts()`, despite the documentation clearly indicating that absolute paths are returned.
 
### Is this a user-facing change? 

- [ ] No. Add the `rn/none` label, then you can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [X] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### Please add one label to the PR so it can be classified correctly in the release notes. Options:
 
* `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
* `rn/none` - No description will be included. The PR will be mentioned just by the PR number in the "Small Bugfixes and Documentation Updates" section
* `rn/feature` - A new user-facing feature worth mentioning in the release notes
* `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
* `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by sueann — APPROVED** [review]
lgtm as long as lint passes - thanks!

File: tests/store/test_local_artifact_repo.py
Comment (sueann) [inline tests/store/test_local_artifact_repo.py]: er, lint does not like this unimported error -- after lint passes i'm good with merging. thanks!

Comment (dbczumar) [reply tests/store/test_local_artifact_repo.py]: Thanks!

**Pull request merged**
